plugins/linuxplugin/ifplugin/data_resync.go

```diff
@@ -33,21 +58,302 @@ func (plugin *LinuxInterfaceConfigurator) Resync(interfaces []*interfaces.LinuxI
+						found = true
+						linuxIf.Veth = &interfaces.LinuxInterfaces_Interface_Veth{
+							PeerIfName: cachedIfCfg.config.HostIfName,
+						}
+					}
+				}
+			}
+			if found {
+				plugin.Log.Debugf("RESYNC Linux interface %v: found peer %v", linkName, linuxIf.Veth.PeerIfName)
+			} else {
+				err := fmt.Errorf("RESYNC Linux interface %v: failed to obtain peer for veth", linkName)
+				errs = append(errs, err)
+			}
+		}
+		// Check if interface needs to be modified
+		if plugin.isLinuxIfModified(linkDataPair.nbIfData, linuxIf) {
+			plugin.Log.Debugf("RESYNC Linux interface %s: configuration changed, interface will be modified", linkName)
+			if err := plugin.ModifyLinuxInterface(linkDataPair.nbIfData, linuxIf); err != nil {
+				errs = append(errs, err)
+			}
+		} else {
+			plugin.Log.Debugf("RESYNC Linux interface %s: data was not changed", linkName)
+		}
 	}
 
 	plugin.Log.WithField("cfg", plugin).Debug("RESYNC Interface end. ", errs)
 
 	return
 }
+
+// Reconstruct common interface configuration from netlink.Link data
+func (plugin *LinuxInterfaceConfigurator) reconstructIfConfig(linuxIf netlink.Link, ns *interfaces.LinuxInterfaces_Interface_Namespace, name string) *interfaces.LinuxInterfaces_Interface {
+	linuxIfAttr := linuxIf.Attrs()
+	return &interfaces.LinuxInterfaces_Interface{
+		Name: name,
+		Type: func(ifType string) interfaces.LinuxInterfaces_InterfaceType {
+			if ifType == veth {
+				return interfaces.LinuxInterfaces_VETH
+			}
+			return interfaces.LinuxInterfaces_AUTO_TAP
+		}(linuxIf.Type()),
+		Enabled: func(state netlink.LinkOperState) bool {
+			if state == netlink.OperDown {
+				return false
+			}
+			return true
+		}(linuxIfAttr.OperState),
+		IpAddresses: plugin.getLinuxInterfaces(linuxIf, ns),
+		PhysAddress: func(hwAddr net.HardwareAddr) (mac string) {
+			if hwAddr != nil {
+				mac = hwAddr.String()
+			}
+			return
+		}(linuxIfAttr.HardwareAddr),
+		Mtu:        uint32(linuxIfAttr.MTU),
+		HostIfName: linuxIfAttr.Name,
+	}
+}
+
+// Reads linux interface IP addresses
+func (plugin *LinuxInterfaceConfigurator) getLinuxInterfaces(linuxIf netlink.Link, ns *interfaces.LinuxInterfaces_Interface_Namespace) (addresses []string) {
+	// Move to proper namespace
+	var revertNs func()
+	if ns != nil {
+		if ns != nil {
+			if !plugin.isNamespaceAvailable(ns) {
+				plugin.Log.Errorf("RESYNC Linux interface %s: namespace is not available", linuxIf.Attrs().Name)
+				return
+			}
+			// Switch to namespace
+			var err error
+			revertNs, err = plugin.switchToNamespace(linuxcalls.NewNamespaceMgmtCtx(), ns)
+			if err != nil {
+				plugin.Log.Errorf("RESYNC Linux interface %s: failed to switch to namespace %s: %v",
+					linuxIf.Attrs().Name, ns.Name, err)
+				return
+			}
+		}
+	}
+
+	// Define defer func to revert namespace if needed
+	defer func(revertNs func()) {
+		if revertNs != nil {
+			revertNs()
+		}
+	}(revertNs)
```

Comment:
You could call `defer revertNs()` after `plugin.switchToNamespace` instead of saving it to other variable and then omit this entire part.